src/njoy21/input/WIMSR/test/WIMSR.test.cpp

```diff
@@ -0,0 +1,313 @@
+    WHEN( "a card2a is expected, but not provided" ){
+      std::string Card1( "-26 27\n" );
+      std::string Card2( "1 4 9/\n" );
+      std::string Card3( "1050 1 1050./\n" );
+      std::string Card4( "3 7 1e10 3 10.890 221 0/\n" );
+      std::string Card7( "1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1./" );
+
+      iRecordStream<char> iss( std::istringstream( 
+        Card1 +
+        Card2 +
+        Card3 +
+        Card4 +
+        Card7 ) );
+
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( WIMSR( iss ) );
+      }
+    } // WHEN
+
+    WHEN( "card5 and card6[a-c] is expected, but not provided" ){
+      std::string Card1( "-26 27\n" );
+      std::string Card2( "1/\n" );
+      std::string Card3( "1050 1 1050. 1/\n" );
+      std::string Card4( "3 7 1e10 3 10.890 221 0/\n" );
+      std::string Card7( "1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1./" );
+
+      iRecordStream<char> iss( std::istringstream( 
+        Card1 +
+        Card2 +
+        Card3 +
+        Card4 +
+        Card7 ) );
+
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( WIMSR( iss ) );
+      }
+    } // WHEN
+
+    WHEN( "one fewer card6c is provided than expected" ){
+      std::string Card1( " -26 27\n" );
+      std::string Card2( " 1 5 9/\n" );
+      std::string Card2a( " 15 4 5 6/\n" );
+      std::string Card3( " 1050 1 1050. 1/\n" );
+      std::string Card4( " 12 13 1.e12 14 0 0 0 1 0 0 2 5/\n" );
+      std::string Card5( " 4 4.8e5/\n" );
+      std::string Card6a( " 1051 1.0/\n" );
+      std::string Card6b( " 1030 0.062/\n" );
+      std::string Card6c( " 1010 0.4/\n" );
+      std::string Card7( " 0.1 0.2 0.3 0.4 0.5/\n" );
+      std::string Card8( " 0.6 0.7 0.8 0.9 1.0/\n" );
+
+      iRecordStream<char> iss( std::istringstream(
+        Card1 +
+        Card2 +
+        Card2a +
+        Card3 +
+        Card4 +
+        Card5 +
+        Card6a +
+        Card6b +
+        Card6c +
+        Card7 +
+        Card8 ) );
+
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( WIMSR( iss ) );
+      }
+    } // WHEN
+
+    WHEN( "Card8 is expected, but not provided" ){
+      std::string Card1( " -26 27\n" );
+      std::string Card2( " 1 5 9/\n" );
+      std::string Card2a( " 15 4 5 6/\n" );
+      std::string Card3( " 1050 1 1050. 1/\n" );
+      std::string Card4( " 12 13 1.e12 14 0 0 0 1 0 0 2 5/\n" );
+      std::string Card5( " 4 4.8e5/\n" );
+      std::string Card6a( " 1051 1.0/\n" );
+      std::string Card6b( " 1030 0.062/\n" );
+      std::string Card6c( " 1010 0.4/\n 1011 0.6/\n" );
+      std::string Card7( " 0.1 0.2 0.3 0.4 0.5/\n" );
+
+      iRecordStream<char> iss( std::istringstream(
+        Card1 +
+        Card2 +
+        Card2a +
+        Card3 +
+        Card4 +
+        Card5 +
+        Card6a +
+        Card6b +
+        Card6c +
+        Card7 ) );
+
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( WIMSR( iss ) );
+      }
+    } // WHEN
+
+/*
```

Comment:
These comments should be removed.

Reply:
I keep wanting to check if extra values/cards are given, but I guess that's more in the scope of the next level up (input.hpp).  Will do.